src/peergos/server/tests/UserTests.java

```diff
@@ -1451,6 +1451,14 @@ public void overwriteContentsOfFileShrinkFile() throws Exception {
                 context.network, context.crypto, len -> {}).join();
 
         checkFileContents(bytes, context.getByPath(filePath).join().get(), context);
+
+        FileWrapper fileV3 = context.getByPath(filePath).join().get();
+        byte[] retrievedData = Serialize.readFully(fileV3.getInputStream(context.network, context.crypto,
+                6000, l-> {}).join(), 6000).join();
+        int nonZeroBytes = (int)IntStream.range(0, retrievedData.length).map(i-> retrievedData[i]).filter(a -> a != 0).count();
+        Assert.assertTrue("File truncated", nonZeroBytes == bytes.length);
```

Comment:
doesn't this rely on all input bytes being non zero?

Reply:
fixed